Add a listing of all projects

tbekolay opened the pull request:
There have been a few instances where it's helpful to have a listing of all the Nengo projects and who maintains them. This PR adds that list.

A few discussion points:

- Are there any other useful pieces of information that we can include for each project?
- Does this organization make sense? Right now projects are grouped as "General", "Backends", "Documentation" and "In progress".

jgosmann commented:
There are some other potentially useful pieces of information that come to my mind, but I wouldn't include them because it would add too much burden to keep this list updated. But maybe (not entirely sure about the usefulness) we could include PyPI version badges (those would update by themselves)?

One question that comes to my mind: How do we decide what projects to include in the list? Things in the nengo GitHub account are pretty obvious, but some backends (i.e. nengo_spinnaker) live under another account. Does that mean we include all backends? Or do we decide on a case by case basis? What about things like [nengolib](https://github.com/arvoelke/nengolib)?

tbekolay commented:
Hm! I like the idea of the PyPI version badges, for repositories that are on PyPI! There might be other package managers involved in the future, so it'd be good to use whichever badge makes sense for the project.

Right yeah, I guess this would be a good place to also list other repositories including nengolib. Is it important to differentiate between "first party" and "third party" repos? I sort of think yes, but not sure the best way to do that... could have a whole separate third party section, or a row in each table though it's not clear what that row would be. "Owner" perhaps? With ABR as the owner of the first-party projects?

jgosmann commented:
> could have a whole separate third party section

I think for a user it is more helpful to have the listing by “General”, “Backend” etc. So a table row might be better. Or maybe some fancy badge indicating first-party projects.

jgosmann commented:
One more thought: `nengo_detailed_neurons` isn't in active development, I would say. @psipeter is using it and as part of that certainly making important progress in how to use detailed neurons, but someone would still have to translate those results in generally usable code, apart from making it easier to install and use `nengo_detailed_neurons` for novice users. Maybe it is useful to indicate such things to? Or maybe that's better done in the project's readme?

Seanny123 commented:
For a diagram of the projects, how do you feel about embedding [the mindmap that I created](https://drive.google.com/open?id=0B3e4kGv2b1RAeXkxeGJyaHNZaTQ).

> Or maybe that's better done in the project's readme?

Yeah, I think that would be the best place for that information.

tbekolay commented:
Here's a pretty big list of badges; there might be more available but it's nice to keep the look the same: https://shields.io/

Note that we can make our own custom badges if we want. If anyone sees any in that list that should be added let me know.

Seanny123 commented on projects.rst:
```diff
@@ -81,128 +108,40 @@ like a particular hardware or software platform,
+   :maintainer: `Daniel Rasmussen <dhrsmss@gmail.com>`_
+
+   Nengo deep learning simulates Nengo models using
+   the `TensorFlow <https://www.tensorflow.org/>`_ library
```
I thought it wast TensorFlow or Lasagne? Has there been a hard switch? When I look at the demo notebook in the repository it's still using Lasagne.

drasmuss replied:
It'll all be switching over to Tensorflow for the first official "release" (maybe in a month or so, rough estimate).

tbekolay commented:
> For a diagram of the projects, how do you feel about embedding the mindmap that I created?

Took another look at this... definitely a good diagram, but I'm not sure it makes sense for this page since it has a bit more information than I was envisioning here... but, it reminded me that I did make a diagram like this a while back for ABR:

![nengo-stack-3d](https://cloud.githubusercontent.com/assets/203709/21823050/904ac512-d747-11e6-9c99-cef40ed8e339.png)

What do you think of this? Probably things should be reorganized a bit and renamed, and we can probably remove the physical hardware row but does this seem useful?

tbekolay commented:
One other thing I was thinking is that the "Maintained by Nengo team" badge looks pretty bad. It would be nice if there was just a Nengo logo which has the tooltip "Managed by Nengo team"; unfortunately there isn't a Nengo logo set in stone right now, but perhaps there should be. I made [this forum thread](https://forum.nengo.ai/t/help-choose-a-new-nengo-logo/199) with a logo idea, would appreciate feedback on it!

Seanny123 commented:
> I'm not sure it makes sense for this page since it has a bit more information than I was envisioning

How about [this reduced diagram](https://drive.google.com/file/d/0B3e4kGv2b1RAX2YzVXREY0J4Zms/view?usp=sharing)? Is this better?

> I did make a diagram like this a while back for ABR

I like that diagram. What did you make it in? Is it possible for me to play around with it?

tbekolay commented:
> What did you make it in? Is it possible for me to play around with it?

It's just an SVG! Made it in Inkscape. Here's the source SVG if you want to make some edits: https://www.dropbox.com/s/r3u13fyrtgmu4ql/nengo-stack-3d.svg?dl=0

tbekolay commented:
I rebased this, added the project interaction diagram, and made it look a little bit nicer so I think I'm happy with how this looks now. At least, it's better than having nothing, so if there are no glaring issues I'll merge this tomorrow morning.

Seanny123 commented:
LGTM.
